seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json

```diff
@@ -2647,6 +2647,48 @@
       }
     },
     {
+      "id": "598e8944f009e646fc236146",
+      "title": "Understanding Undefined Value returned from a Function",
+      "description": [
+        "A function can include the <code>return</code> statement but it does not have to. In case that the function does not have a <code>return</code> statement, the function will process the inner code and the returned value in that case will be <code>undefined</code>.",
+        "<strong>Example</strong>",
+        "<blockquote>var sum = 0; function addSum(num) {<br>  sum = sum + num;<br>}<br>var returnedValue = addSum(3); // sum will be modified but returned value is undefined</blockquote>",
+        "<code>addSum</code> is a function without a <code>return</code> statement. The function will change the global <code>sum</code> variable but the returned value of the function is <code>undefined</code>",
+        "<hr>",
+        "Create a function <code>addFive</code> without any argument. This function add 5 to the <code>sum</code> variable, but its returned value is <code>undefined</code>."
+      ],
+      "releasedOn": "August 11, 2017",
+      "challengeSeed": [
+        "// Example",
+        "var sum = 0;",
+        "function addThree() {",
+        "  sum = sum + 3;",
+        "}",
+        "",
+        "// Only change code below this line",
+        "",
+        "",
+        "",
+        "// Only change code above this line.",
+        "var returnedValue = addFive();"
+      ],
+      "tail": [
+        "var sum = 0;",
+        "addFive();"
+      ],
+      "solutions": [
+        "function addFive() {\n sum = sum + 5;\n}"
+      ],
+      "tests": [
+        "assert(typeof addFive === 'function', 'message: <code>addFive</code> should be a function');",
+        "assert(sum === 5, 'message: <code>sum</code> should contain value 5');",
+        "assert(addFive() === undefined, 'message: returned value from <code>addFive</code> should be <code>undefined</code>');"
```

Comment:
I think campers may be confused with this test, because they'll see the code above theirs that adds 3 to `sum`, then their function, and will likely expect the result to be 8. We should change the tail and this test to check for 8 instead, to keep things transparent.